packages/headless/src/controllers/recent-queries-list/headless-recent-queries-list.ts

```diff
@@ -0,0 +1,169 @@
+import {ArrayValue, NumberValue, Schema} from '@coveo/bueno';
+import {buildController, Controller} from '../controller/headless-controller';
+import {search, recentQueries} from '../../app/reducers';
+import {loadReducerError} from '../../utils/errors';
+import {SearchEngine} from '../../app/search-engine/search-engine';
+import {RecentQueriesSection} from '../../state/state-sections';
+import {RecentQueriesState} from '../../features/recent-queries/recent-queries-state';
+import {
+  clearRecentQueries,
+  registerRecentQueries,
+} from '../../features/recent-queries/recent-queries-actions';
+import {
+  validateInitialState,
+  validateOptions,
+} from '../../utils/validate-payload';
+import {executeSearch} from '../../features/search/search-actions';
+import {updateQuery} from '../../features/query/query-actions';
+import {
+  logClearRecentQueries,
+  logRecentQueryClick,
+} from '../../features/recent-queries/recent-queries-analytics-actions';
+
+export interface RecentQueriesListProps {
+  /**
+   * The initial state that should be applied to the `RecentQueriesList` controller.
+   */
+  initialState?: RecentQueriesListInitialState;
+  /**
+   * The configuration options that should be applied to the `RecentQueriesList` controller.
+   */
+  options?: RecentQueriesListOptions;
+}
+
+const defaultRecentQueriesProps: Required<RecentQueriesListProps> = {
+  initialState: {
+    queries: [],
+  },
+  options: {
+    maxLength: 10,
+  },
+};
+
+export interface RecentQueriesListInitialState {
+  /**
+   * The list of recent queries
+   * @defaultValue `[]`
+   */
+  queries: string[];
```

Comment:
Since this has a default value and has `required: false` in its validation schema, I think it should be optional.
```suggestion
  queries?: string[];
```

Reply:
I guess it should be required: true then. We don't want it to be possible de pass an empty object as options right?